dts: riscv: define riscv,machine-timer

gmarull opened the pull request:
All RISC-V platforms with machine timer, except GigaDevice, define its
base address in soc.h. Nowadays we have Devicetree to describe hardware,
so it needs to be used when appropriate. This patch defines the machine
timer for all platforms that define RISCV_MTIME_BASE and
RISCV_MTIMECMP_BASE in soc.h, re-using the same base addresses.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
dts: riscv: define riscv,machine-timer

All RISC-V platforms with machine timer, except GigaDevice, define its
base address in soc.h. Nowadays we have Devicetree to describe hardware,
so it needs to be used when appropriate. This patch defines the machine
timer for all platforms that define RISCV_MTIME_BASE and
RISCV_MTIMECMP_BASE in soc.h, re-using the same base addresses.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: timer: riscv_machine_timer: obtain base address/IRQ from DT

Now that all platforms define the timer in DT, obtain its base address
and IRQ from DT.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
soc: riscv: remove unused RISCV_MTIME(CMP)_BASE and IRQ definitions

The RISC-V machine timer base addresses and IRQ are now obtained from
Devicetree.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
soc: riscv: mpfs: remove unused definition

RISCV_MTIMECMP_BY_HART (machine timer related) is not used, so remove
it.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
dts: riscv: gigadevice: fix riscv,machine-timer reg length

Machine timer register are 64-bit, so their length is 8 bytes.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

galak reviewed (changes requested):
Can you update the binding to describe the 2 register regions in the 'reg' property description. (And make `reg` required).

Wonder about adding `reg-names` as well??

gmarull committed:
dts: bindings: require reg/reg-names/interrupts and improve description

Improve the description of the binding by providing more context to the
user, together with an example usage. Also make reg, reg-names and
interrupts required.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
dts: riscv: gigadevice: add missing reg-names to machine timer

Names will be required by the machine timer driver, update dts.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented:
> Can you update the binding to describe the 2 register regions in the 'reg' property description. (And make `reg` required).
> 
> Wonder about adding `reg-names` as well??

done, using names makes things much better indeed.

galak commented on dts/bindings/timer/riscv,machine-timer.yaml:
```diff
@@ -1,13 +1,35 @@
+    reg:
+      required: true
+
+    reg-names:
```
Add in `description` what the `reg-names` values are.

gmarull replied:
There's an example in the description. Is it a good practice to override descriptions from fields that already do? This is a generic property already documented in `base.yaml`.

gmarull committed:
dts: riscv: gigadevice: add missing interrupt for machine timer

Machine timer did not have interrupt defined, add it.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

carlocaione commented on dts/riscv/telink/telink_b91.dtsi:
```diff
@@ -40,6 +40,15 @@
+			compatible = "riscv,machine-timer";
+			reg = <0xe6000000 0x8
+			       0xe6000008 0x8>;
+			interrupts = <7 0>;
```
what the interrupt `0` is for? That should be the `User software interrupt` but is that used anywhere?

gmarull replied:
`0` is the priority, unless this platform does it differently

carlocaione replied:
hahaha, of course. I'll go hiding in shame, see ya.
